src/Model/Item.php

```diff
@@ -2848,12 +2848,12 @@ private static function samePermissions($obj1, $obj2)
 	}
 
 	// returns an array of contact-ids that are allowed to see this object
-	public static function enumeratePermissions($obj)
+	public static function enumeratePermissions($uid, array $obj)
```

Comment:
Is there a reason adding the uid to the function when it is already part of the item array?

Reply:
It looks like we can use the `uid` of the provided array, either in the `item` context or in the `photo` context.